sotodlib/site_pipeline/make_atomic_filterbin_map.py

```diff
@@ -0,0 +1,556 @@
+                # we are asking for splits
+                missing_split = False
+                for split_label in split_labels:
+                    query_ = 'SELECT * from atomic where obs_id="%s" and telescope="%s" and freq_channel="%s" and wafer="%s" and split_label="%s"'%(value[0][0], obs_infos[value[0][3]].telescope, key[2], key[1], split_label )
+                    res = cursor.execute(query_)
+                    matches = res.fetchall()
+                    if len(matches)==0:
+                        # this means one of the requested splits is missing
+                        # in the data base
+                        missing_split = True
+                        break
+                if missing_split == False:
+                    # this means we have all the splits we requested for the
+                    # particular obs_id/telescope/freq/wafer
+                    keys_to_remove.append(key)
+        for key in keys_to_remove:
+            obskeys.remove(key)
+            del obslists[key]
+        conn.close() # I close since I only wanted to read
+
+    obslists_arr = [item for key, item in obslists.items()]
+    my_oblists=[]; my_tods = []; my_ra_ref=[]; pwvs=[]
+    L.info('Starting with read_tods')
+    with ProcessPoolExecutor(args['nproc']) as exe:
+        futures = [exe.submit(read_tods, args['context'], obslist, 
+                             dtype_tod=args['dtype_tod'],
+                              only_hits=args['only_hits'],
+                              l2_data=args['l2_data_path']) for obslist in obslists_arr]
+        for future in as_completed(futures):
+            #L.info('New future as_completed result')
+            try:
+                my_obslist_here, my_tods_here, my_ra_ref_here, pwvs_here = future.result()
+                my_oblists.append(my_obslist_here)
+                my_tods.append(my_tods_here)
+                my_ra_ref.append(my_ra_ref_here)
+                pwvs.append(pwvs_here)
+            except Exception as e:
+                errmsg = f'{type(e)}: {e}'
+                tb = ''.join(traceback.format_tb(e.__traceback__))
+                L.info(f"ERROR: future.result()\n{errmsg}\n{tb}")
+                f = open(errlog, 'a')
+                f.write(f'\n{time.time()}, future.result() error\n{errmsg}\n{tb}\n')
+                f.close()
+                continue
+            futures.remove(future)
+    # flatten the list of lists
+    my_oblists = list(itertools.chain.from_iterable(my_oblists))
+    my_tods = list(itertools.chain.from_iterable(my_tods))
+    my_ra_ref = list(itertools.chain.from_iterable(my_ra_ref))
+    pwvs = list(itertools.chain.from_iterable(pwvs))
+    L.info('Done with read_tods')
+
+    if args['area'] is not None:
+        # we will do the profile and footprint here, and then allgather the
+        # subshapes and subwcs.This way we don't have to communicate the
+        # massive arrays such as timestamps
+        subshapes = [] ; subwcses = []
```

Comment:
A single list, called "subgeoms", containing (shape, wcs) tuples, would be much easier to work with.  Later you can just do `subshape, subwcs = subgeoms[oi]`.